src/components/home/TopicDrawer.js

```diff
@@ -0,0 +1,163 @@
+import Typography from '@material-ui/core/Typography';
+
+import { searchEnvironmentsByTopic, resetSearchedEnvironments } from '../../actions/environments';
+import About from './About';
+
+const drawerWidth = 240;
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+    backgroundColor: 'AliceBlue',
+  },
+  title: {
+    margin: theme.spacing.unit * 2,
+    marginTop: theme.spacing.unit * 6,
+  },
+  drawerPaper: {
+    width: drawerWidth,
+    flexShrink: 0,
+    paddingTop: '0',
+    paddingBottom: '0',
+  },
+  wrapper: {
+    display: 'flex',
+    flexFlow: 'row nowrap',
+  },
+  gridStyle: {
+    [theme.breakpoints.up('lg')]: {
+      width: '100%',
+    },
+  },
+});
+
+class TopicDrawer extends Component {
+  constructor(props) {
+    super(props);
+    const topTopics = this.props.topics.filter(topic => !topic.parentTopic);
+    this.state = {
+      open: Array.apply(null, Array(topTopics.length)).map(function() {
+        return false;
+      }),
+      error: '',
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.topics !== prevProps.topics) {
+      const topTopics = this.props.topics.filter(topic => !topic.parentTopic);
+      this.setState({
+        open: Array.apply(null, Array(topTopics.length)).map(function() {
+          return false;
+        }),
+      });
+    }
+  }
+
+  handleTopClick = (index, id) => {
+    //TODO: handle FAILURE
+    const openList = this.state.open;
+    openList[index] = !openList[index];
+    this.setState({ open: openList });
+    this.props.searchEnvironmentsByTopic(id);
+  };
+
+  handleClick = id => {
+    //TODO: handle FAILURE
+    if (id === 'all') {
+      this.props.resetSearchedEnvironments();
+    } else this.props.searchEnvironmentsByTopic(id);
+  };
+
+  render() {
+    const { classes, topics } = this.props;
+
+    const topTopics = topics.filter(topic => !topic.parentTopic);
+    const botTopics = topics.filter(topic => topic.parentTopic);
+    return (
+      <Hidden mdDown>
+        <Paper className={classes.drawerPaper}>
+          <About />
+          <Typography variant="h5" className={classes.title}>
+            Search Categories
+          </Typography>
+          <MenuList>
+            <ListItem button key={'all'} onClick={() => this.handleClick('all')}>
+              <ListItemText primary={'Recent Environments'} />
+            </ListItem>
+            <Divider />
+            {topTopics.map((topTopic, index) => (
+              <div key={index}>
+                <ListItem
+                  button
+                  key={topTopic.id}
+                  onClick={() => this.handleTopClick(index, topTopic.id)}
+                >
+                  <ListItemText primary={topTopic.name} />
+                  {this.state.open[index] ? <ExpandLess /> : <ExpandMore />}
+                </ListItem>
+                <Collapse in={this.state.open[index]} timeout="auto" unmountOnExit>
+                  {botTopics.map(
```

Comment:
When a `render` function gets this long it's usually a sign that it should be broken up into different components. For example the JSX in these nested map function could be moved to it's own component.